mesonbuild/utils/universal.py

```diff
@@ -472,7 +472,7 @@ def get_compiler_for_source(compilers: T.Iterable['Compiler'], src: 'FileOrStrin
     raise MesonException(f'No specified compiler can handle file {src!s}')
 
 
-def classify_unity_sources(compilers: T.Iterable['Compiler'], sources: T.Sequence['FileOrString']) -> T.Dict['Compiler', T.List['FileOrString']]:
+def classify_unity_sources(compilers: T.Iterable['Compiler'], sources: T.Union[T.List['FileOrString'], T.List[File]]) -> T.Dict['Compiler', T.List['FileOrString']]:
```

Comment:
Can I interest you in something much better: 3ca39913364cce287399cf07c4d194221788d403?

This has the advantage of knowing that if you give it a `list[str]` you get `list[str]`, etc. Using this makes a lot of the Union work in your next patch unnecessary, since mypy can know up front that it's passing a non-union type.

Reply:
(or maybe it was other uncommitted cases of the same issue? Not sure, brain fried and can't check. :D)

Reply:
Now that I look again, your patch reiterates the exact issue that I was trying to fix. If you had correctly typed the function using any method, whether expanding covariance out via Union or via generics, ***you would know you got it correct*** -- because it would reveal an existing error in backend.py

It feels to me like you didn't actually read my commit message. :(

Reply:
I'll leave things alone until you return, so please leave this alone until then and we'll discuss this further at that point. Accept my apologies for stepping on your toes here, that wasn't my intention. When you have finished with your holy days, we can discuss this further.